vms/platformvm/state/chain_time_helpers.go

```diff
@@ -64,42 +75,83 @@ func GetNextStakerChangeTime(state Chain, defaultTime time.Time) (time.Time, err
 		}
 
 		time := it.Value().NextTime
-		if time.Before(defaultTime) {
-			defaultTime = time
+		if time.Before(nextTime) {
+			nextTime = time
 		}
 	}
-	return defaultTime, nil
+
+	sovIterator, err := state.GetActiveSubnetOnlyValidatorsIterator()
```

Comment:
I moved the code into a separate function because I feel like it increases readability... But I feel like the existing tests for this logic are more comprehensive and easier to write than a test for the new helper.